gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/CategoryServiceImpl.java

```diff
@@ -256,7 +260,9 @@ public List<CategoryEntity> update(ExecutionContext executionContext, final List
     @Override
     public void delete(ExecutionContext executionContext, final String categoryId) {
         try {
-            Optional<Category> categoryOptional = categoryRepository.findById(categoryId);
+            Optional<Category> categoryOptional = categoryRepository
+                .findById(categoryId)
+                .filter(category -> category.getEnvironmentId().equalsIgnoreCase(executionContext.getEnvironmentId()));
```

Comment:
orElseGet ?

Reply:
No because we do nothing if we don't find the category.